src/Symfony/Component/Console/Attribute/Argument.php

```diff
@@ -41,10 +43,12 @@ class Argument
      * @param array<string|Suggestion>|callable(CompletionInput):list<string|Suggestion> $suggestedValues The values used for input completion
      */
     public function __construct(
-        public string $description = '',
+        string|callable $description = '',
         public string $name = '',
         array|callable $suggestedValues = [],
     ) {
+        $this->description = \is_callable($description) ? $description() : $description;
```

Comment:
I would apply the same approach we used for suggestedValues, so that non-static callables are supported. This way, we can access the `$this` instance and its dependencies if needed, for example, to build a description dynamically based on the command's dependencies

Reply:
Alright, I've tried to do that, but I've noticed a problem: I don't think I can do that without breaking compatibility.

As for now, `$description` is a `public string` field: https://github.com/symfony/symfony/blob/v7.3.4/src/Symfony/Component/Console/Attribute/Argument.php#L39

I'd have to change it to `public string|\Closure`, so if someone's code relies upon `$description` being a string, it would break it.